src/util/sighash.rs

```diff
@@ -671,8 +679,11 @@ impl<R: Deref<Target = Transaction>> SighashCache<R> {
         sighash_type: u32,
     ) -> Result<Sighash, Error> {
         let mut enc = Sighash::engine();
-        self.legacy_encode_signing_data_to(&mut enc, input_index, script_pubkey, sighash_type)?;
```

Comment:
Was not handling this case a mistake?

Reply:
Looks like it, was probably me that missed it. Good that you caught it.